### ISSUE TITLE
/r control character is causing client issues

### ISSUE THREAD

**Issue by juergh** [bstansell/conserver#40]
I'm new to conserver so I must be doing something wrong.

I have conserver connected to a couple of serial consoles. It seems that the '/r' control character used in process bars is messing up  my clients. By that I mean that scrolling no longer works properly and only the last line of the terminal is updated. If that makes any sense :-) I then have to disconnect, reset the terminal and reconnect.

Is this a config issue on the server? Or do I need to supply some argument to the client?

**Comment by conserver** [comment]
Yeah, there are multiple ways this can go sideways.  I doubt it's a conserver issue - it pretty much just pipes things to the client sessions and out to the user.  Depending on how you define your consoles and what happens when things connect (like terminal settings [see stty]), things could be told to "eat" the `\r` - on both the server and client side.  Without a serious amount of detail of your setup, I'm not sure what to suggest.  As a first pass, I'd suggest checking stty output on the near and far end and compare with any working setup to see if you can find a difference.  That might highlight why/how things are not being processed.

**Comment by juergh** [comment]
It seems it's not strictly related to '\r' but pagers seem to cause problems too (using man <foo> for example). But only after I resize the client terminal. A terminal with != 24 lines is not a happy one, even after disconnect/reset/reconnect.

**Comment by bstansell** [comment]
Yeah, this is a common "problem".  Anything attached to a pseudo-terminal is going to expect a certain sized screen...and by default it sounds like yours expects 24 rows (and possibly 80 columns).  Running `stty -a` will show you what it expects and you can set it's view of the world using the same tool.  The `resize` X11 command is useful in that it will send escape sequences your terminal hopefully understands to query the current size and sets things appropriately.

Bottom line: this is a classic "my environment doesn't match the remote console's expected view of the world".
